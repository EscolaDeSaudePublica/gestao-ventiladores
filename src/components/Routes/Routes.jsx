import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import { randomIndex } from '../../utils';
import IndexServiceOrder from '../../pages/SerrviceOrder';

export const hasRoutes = (routes) => {
  if (routes && routes.length > 0) {
    return true;
  }
  return false;
};

const Routes = ({ routes }) => (
  <BrowserRouter>
    <Switch>
      {hasRoutes(routes) ? routes.map((route) => (
        <Route key={randomIndex()} path={route.path} exact component={route.component} />
      )) : <Route key={randomIndex()} path="/" exact component={IndexServiceOrder} />}
    </Switch>
  </BrowserRouter>
);

Routes.propTypes = {
  routes: PropTypes.instanceOf(Array).isRequired,
};

export default Routes;
