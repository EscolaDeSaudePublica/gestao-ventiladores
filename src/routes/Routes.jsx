import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import { randomIndex } from '../utils';

const RequireRotes = () => <h1>Routes component require rotes as props</h1>;

export const hasRoutes = (routes) => {
  if (routes && routes.length > 0) {
    return true;
  }
  return false;
};

const getRoutes = (routes) => {
  if (hasRoutes(routes)) {
    return routes.map((route) => (
      <Route
        key={randomIndex()}
        path={route.path}
        exact={route.exact}
        component={route.component}
      />
    ));
  }
  return <Route key={randomIndex()} path="/" exact component={RequireRotes} />;
};

const Routes = ({ routes }) => (
  <BrowserRouter>
    <Switch>{getRoutes(routes)}</Switch>
  </BrowserRouter>
);

Routes.propTypes = {
  routes: PropTypes.instanceOf(Array).isRequired,
};

export default Routes;
