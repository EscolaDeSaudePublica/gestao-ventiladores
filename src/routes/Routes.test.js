import React from 'react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Routes, { hasRoutes } from './Routes';
import { render } from '@testing-library/react';

describe('Routes component', () => {
  describe('hasRoutes function', () => {
    it('should false if undefined routes', () => {
      expect(hasRoutes()).toBe(false);
    });

    it('should false if routes is empty array', () => {
      expect(hasRoutes([])).toBe(false);
    });
  });

  it('should throws error when there is no routes props', () => {
    const error = new Error('Routes component require routes as props');
    expect(() => render(<Routes />)).toThrow(error);
  });

  it('should throws error when routes is empty array', () => {
    const error = new Error('Routes component require routes as props');
    expect(() => render(<Routes routes={[]} />)).toThrow(error);
  });

  it('should display component without errors when path redirect to valid route', () => {
    const Todo = () => (
      <>
        <p>To do</p>
      </>
    );

    const routes = [{ path: '/', component: Todo }];

    const component = create(<Routes routes={routes} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
