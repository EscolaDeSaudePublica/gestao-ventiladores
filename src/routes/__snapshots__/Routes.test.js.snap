// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component should display component without errors when no routes props 1`] = `
<h1>
  Routes component require rotes as props
</h1>
`;

exports[`Routes component should display component without errors when path redirect to valid route 1`] = `
<p>
  To do
</p>
`;

exports[`Routes component should display component without errors when routes is empty array 1`] = `
<h1>
  Routes component require rotes as props
</h1>
`;
